softwareComponents/torqueComputation/src/traverseRofiWorld.cpp

```diff
@@ -0,0 +1,266 @@
+) {
+        std::unordered_map<std::pair<int, int>, int, IntPairHash> result;
+        for (const auto& roficom : world.roficomConnections()) {
+
+            const auto& fromModule = roficom.getSourceModule(world);
+            int fromCompId = roficom.sourceConnector;
+
+            const auto& toModule = roficom.getDestModule(world);
+            int toCompId = roficom.destConnector;
+
+            int jointId = getJointId(fromModule.getId(), fromCompId);
+
+            result[std::make_pair(fromModule.getId(), fromCompId)] = jointId;
+            result[std::make_pair(toModule.getId(), toCompId)] = jointId;
+
+            auto roficomJoint = std::unique_ptr<Joint>(new Joint(
+                jointId,
+                getRoficomPosition(fromModule.components()[fromCompId]),
+                false
+            ));
+            joints.emplace(
+                jointId, 
+                std::move(roficomJoint)
+            );
+        }      
+
+        return result;  
+}
+
+void VisitPadModuleStrategy::createJoints(
+        const rofi::configuration::Module& rofiModule,
+        const std::unordered_map<std::pair<int, int>, int, IntPairHash>& roficomMap,
+        std::unordered_map<int, std::unique_ptr<Joint>>& joints,
+        VariableManager&,
+        const arma::vec& gravitation
+) const {
+    int moduleId = rofiModule.getId();
+    for (const auto& padComponent : rofiModule.components()) {
+        auto padJointIdx = roficomMap.find(std::make_pair(moduleId, padComponent.getIndexInParent()));
+
+        if (padJointIdx != roficomMap.end()) {
+            //Roficom already created during traversing roficom connections
+            auto* padJoint = joints.at(padJointIdx->second).get();
+            padJoint->appendForce(getConfiguration().weights[ComponentTypeIndex::Roficom] * gravitation);
+            padJoint->setIsWall();
+        }
+    }
+}
+
+void VisitUniversalModuleStrategy::createJoints(
+        const rofi::configuration::Module& rofiModule,
+        const std::unordered_map<std::pair<int, int>, int, IntPairHash>& roficomMap,
+        std::unordered_map<int, std::unique_ptr<Joint>>& joints,
+        VariableManager& variableManager,
+        const arma::vec& gravitation
+) const {
+    const auto& shoeA = rofiModule.components()[6];
+    const auto& shoeB = rofiModule.components()[9];
+    arma::vec delta = (getPosition(shoeB) - getPosition(shoeA)) / 3;
+    arma::vec middle = (getPosition(shoeB) + getPosition(shoeA)) / 2;
+    const auto& configuration = getConfiguration();
+    int multiplier = 1;
+
+    for (const auto& cmpJoint : rofiModule.joints()) {
+        auto visitor = rofi::configuration::JointVisitor::make(
+            [
+                &rofiModule,
+                &joints,
+                &variableManager,
+                &gravitation,
+                &cmpJoint,
+                &delta,  
+                &configuration,
+                &multiplier
```

Comment:
Celá lambda je hodně dlouhá a nepřehledná. Myslím, že explicitní vyjmenování capture nepřidává žádnou informaci navíc a klidně bych to jen nahradil za `[&]`